src/libData/AccountData/AccountStore.cpp

```diff
@@ -288,7 +375,9 @@ bool AccountStore::UpdateAccountsTemp(const uint64_t& blockNum,
                                       TransactionReceipt& receipt) {
   // LOG_MARKER();
 
-  lock_guard<mutex> g(m_mutexDelta);
+  unique_lock<shared_timed_mutex> g(m_mutexPrimary, defer_lock);
+  unique_lock<mutex> g2(m_mutexDelta, defer_lock);
```

Comment:
Why perform 2 different kinds of mutex here?

Reply:
Because the m_mutexPrimary protects the state trie, the m_mutexDelta protects the AccountStoreTemp for state delta